File: API/Entities/OrderAggregate/Order.cs
namespace API.Entities.OrderAggregate;

public class Order
{
    public int Id { get; set; }
    public required string BuyerEmail { get; set; }
    public required ShippingAddress ShippingAddress { get; set; }
    public DateTime OrderDate { get; set; } = DateTime.UtcNow;
    public required List<OrderedItem> OrderedItems { get; set; } = [];
    public long Subtotal { get; set; }
    public long DeliveryFee { get; set; }
    public required string PaymentIntentId { get; set; }
    public OrderStatus OrderStatus { get; set; } = OrderStatus.Pending;
    public required PaymentSummary PaymentSummary { get; set; }

    public long GetTotal()
    {
        return Subtotal + DeliveryFee;
    }
}

File: client/src/models/Order.ts
export type Order = {
  id: number
  buyerEmail: string
  shippingAddress: ShippingAddress
  orderDate: string
  orderedItems: OrderedItem[]
  subtotal: number
  deliveryFee: number
  total: number
  orderStatus: string
  paymentSummary: PaymentSummary
}

export type ShippingAddress = {
  name: string
  line1: string
  line2?: string | null
  city: string
  state: string
  postal_code: string
  country: string
}

export type OrderedItem = {
  productId: number
  name: string
  pictureUrl: string
  price: number
  quantity: number
}

export type PaymentSummary = {
  last4: number | string
  brand: string
  exp_month: number
  exp_year: number
}

export type CreateOrder = {
    shippingAddress: ShippingAddress
    paymentSummary: PaymentSummary
}
